tests/integration_tests/cmd/test_status.py

```diff
@@ -0,0 +1,58 @@
+"""Tests for `cloud-init status`"""
+from time import sleep
+
+import pytest
+
+from tests.integration_tests.clouds import ImageSpecification, IntegrationCloud
+from tests.integration_tests.instances import IntegrationInstance
+
+
+# We're implementing our own here in case cloud-init status --wait
+# isn't working correctly (LP: #1966085)
+def _wait_for_cloud_init(client: IntegrationInstance):
+    last_exception = None
+    for _ in range(30):
+        try:
+            result = client.execute("cloud-init status")
+            if result and result.ok:
+                return result
+        except Exception as e:
+            last_exception = e
+        sleep(1)
+    raise Exception(
+        "cloud-init status did not return successfully."
+    ) from last_exception
+
+
+def _remove_nocloud_dir_and_reboot(client: IntegrationInstance):
+    client.execute("rm -rf /var/lib/cloud/seed/nocloud-net")
+    client.execute("cloud-init clean --logs --reboot")
+
+
+@pytest.mark.ubuntu
+@pytest.mark.lxd_container
+def test_wait_when_no_datasource(session_cloud: IntegrationCloud, setup_image):
+    """Ensure that when no datasource is found, we get status: disabled
+
+    LP: #1966085
+    """
+    with session_cloud.launch(
+        launch_kwargs={
+            "config_dict": {"security.devlxd": False},
```

Comment:
I don't think I understand this bit. Are these args the reason that no datasource is found?

Reply:
I added some comments. Do they answer your question?

Reply:
Perfect, thanks!